### PR TITLE
Add numWrittenFiles in RuntimeStats for TableWriter

### PR DESCRIPTION
**Pull request by kewang1024**

### CODE REVIEW

File: velox/connectors/Connector.h [hunk 1]
```diff
@@ -146,6 +146,10 @@ class DataSink {
     return 0;
   }
 
+  virtual int32_t getNumWrittenFiles() const {
```
Comment (xiaoxmeng) [inline velox/connectors/Connector.h:149]:
```
s/getNumWrittenFiles/numWrittenFiles/
```
If there is no other connector implementation, can we just throw with VELOX_NYI?

File: velox/exec/tests/TableWriteTest.cpp [hunk 2]
```diff
@@ -2743,6 +2743,7 @@ TEST_P(AllTableWriterTest, tableWrittenBytes) {
   for (int i = 0; i < operatorStats.size(); i++) {
     if (operatorStats.at(i).operatorType == "TableWrite") {
       ASSERT_GT(operatorStats.at(i).physicalWrittenBytes, 0);
+      ASSERT_GT(operatorStats.at(i).runtimeStats.at("numWrittenFiles").sum, 0);
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2746]: Can we literally check the number of written files? Thanks!

File: velox/exec/TableWriter.cpp [hunk 3]
```diff
@@ -162,6 +162,9 @@ RowVectorPtr TableWriter::getOutput() {
 
   finished_ = true;
   updateWrittenBytes();
```
Comment (xiaoxmeng) [inline velox/exec/TableWriter.cpp:164]: Can we update inside updateWrittenBytes? so we grab the lock once? We can rename s/updateWrittenBytes/updateWrittenStats/

**Review by xiaoxmeng — COMMENTED** [review]
@kewang1024 LGTM. Thanks % minors!

File: velox/connectors/Connector.h (hunk 1 above)
Comment (kewang1024) [reply velox/connectors/Connector.h:149]: The function above is called `getCompletedBytes`, so I will keep the name to stay consistent.

Comment (xiaoxmeng) [reply velox/connectors/Connector.h:149]: Let's remove get prefix for the new method? Thanks!

File: velox/exec/tests/TableWriteTest.cpp [hunk 4]
```diff
@@ -2739,10 +2739,21 @@ TEST_P(AllTableWriterTest, tableWrittenBytes) {
...
       plan, inputFilePaths, "SELECT count(*) FROM tmp");
-  auto operatorStats = task->taskStats().pipelineStats.at(0).operatorStats;
-  for (int i = 0; i < operatorStats.size(); i++) {
-    if (operatorStats.at(i).operatorType == "TableWrite") {
-      ASSERT_GT(operatorStats.at(i).physicalWrittenBytes, 0);
+
+  // Each batch would create a new partition, numWrittenFiles is same as
+  // partition num when not bucketed. When bucketed, it's partitionNum *
+  // bucketNum, bucket number is 4
+  int numWrittenFiles =
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2746]:
```
const int numWrittenFiles =
```

**Review by xiaoxmeng — APPROVED** [review]
@kewang1024 thanks for the update!

**Pull request closed without merging**